sdk/scheduler/src/main/java/org/apache/mesos/specification/DefaultServiceSpecification.java

```diff
@@ -29,19 +33,48 @@
     public static class Factory implements ConfigurationFactory<ServiceSpecification> {
 
         /**
-         * Call {@link DefaultServiceSpecification#getFactoryInstance()} instead.
+         * Subtypes to be registered by defaults. This list should include all
+         * {@link PlacementRule}s that are included in the library.
          */
-        private Factory() { }
+        private static final Collection<Class<?>> defaultRegisteredSubtypes = Arrays.asList(
+                AgentRule.class,
+                AndRule.class,
+                AttributeRule.class,
+                HostnameRule.class,
+                MaxPerAttributeRule.class,
+                NotRule.class,
+                OrRule.class,
+                TaskTypeRule.class);
```

Comment:
In their types they need to:
- Ensure that they specify `@JsonCreator` and `@JsonProperty` in their constructor
- Ensure that their members are accessible via 'get' methods with `@JsonProperty` annotations.

In the Scheduler they need to:
- Add the Classes of their custom types into DefaultServiceSpecification.Factory() at the time of constructing the DefaultScheduler:
```
DefaultScheduler.create(
    serviceSpecification,
    DefaultScheduler.createStateStore(
        serviceSpecification.getName(),
        DcosConstants.MESOS_MASTER_ZK_CONNECTION_STRING),
    DefaultScheduler.createConfigStore(
        serviceSpecification.getName(),
        DcosConstants.MESOS_MASTER_ZK_CONNECTION_STRING),
        Arrays.asList(Custom.class, Types.class, Here.class)));
```